src/Featurizers/LagLeadOperatorFeaturizer.h

```diff
@@ -0,0 +1,224 @@
+// ----------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License
+// ----------------------------------------------------------------------
+#pragma once
+
+#include "../Featurizer.h"
+#include "../Archive.h"
+#include "Components/PipelineExecutionEstimatorImpl.h"
+#include "Components/FilterDecoratorFeaturizer.h"
+#include "Components/GrainFeaturizerImpl.h"
+#include "Components/WindowFeaturizerBase.h"
+
+namespace Microsoft {
+namespace Featurizer {
+namespace Featurizers {
+
+/////////////////////////////////////////////////////////////////////////
+///  \class         LagLeadOperatorTransformer
+///  \brief         Copy values from prior data or future data
+///                 Input type is a tuple of std::vector<std::string>, representing grains, and InputT, representing target column
+///
+template <typename InputT>
+class LagLeadOperatorTransformer :
+    public Transformer<
+        InputT,
+        Microsoft::Featurizer::RowMajMatrix<nonstd::optional<InputT>>
```

Comment:
>nonstd::optional<InputT> [](start = 44, length = 24)

use `typename Traits<InputT>::nullable_type`, as an `InputT` of `double` should not be wrapped in `nonstd::optional`.

Reply:
Good point. Changed to nullable_type.